src/carLogic/carLogic.ino

```diff
@@ -191,3 +193,43 @@ void shake(int speed) {
   
   car.setSpeed(0);
 }
+
+void macarena(int speed) {
+  long startingPoint = 0;
+  int steps = 1;
+  int repeats = 0;
+
+  while (repeats != 3) {
+    if (steps == 1) {
+      startingPoint = car.getDistance();
+      car.setSpeed(speed);
+      steps++;
+    } else if ((steps == 2 || steps == 4) && (car.getDistance() - startingPoint) == 15) {
+      car.setSpeed(0);
+      delay(1000);
+      car.setSpeed(speed * -1);
+      steps++;
+    } else if ((steps == 3) && (car.getDistance() - startingPoint == 0)){
+      car.setSpeed(0);
+      delay(1000);
+      car.setSpeed(speed);
+      steps++;
+    } else if ((steps == 5) && (car.getDistance() - startingPoint == 0)) {
+     car.setSpeed(0);
+      delay(1000);
+      rotateOnSpot(90,100);
+      repeats++;
+      
+    }
+    /*car.setSpeed(0);
```

Comment:
Remove all commented out code and all extra lines from line 231 to the last bracket. :)